test/common/event/user_space_file_event_impl_test.cc

```diff
@@ -0,0 +1,210 @@
+#include <cstdint>
+
+#include "envoy/event/file_event.h"
+
+#include "common/event/dispatcher_impl.h"
+#include "common/event/user_space_file_event_impl.h"
+
+#include "test/mocks/common.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/test_runtime.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Event {
+namespace {
+
+constexpr auto event_rw = Event::FileReadyType::Read | Event::FileReadyType::Write;
+class MockReadyCb {
+public:
+  MOCK_METHOD(void, called, (uint32_t));
+};
+
+class UserSpaceFileEventImplTest : public testing::Test {
+public:
+  UserSpaceFileEventImplTest()
+      : api_(Api::createApiForTest()), dispatcher_(api_->allocateDispatcher("test_thread")) {
+    io_callback_ =
+        dispatcher_->createSchedulableCallback([this]() { user_file_event_->onEvents(); });
+  }
+
+  void scheduleNextEvent() {
+    ASSERT(io_callback_ != nullptr);
+    io_callback_->scheduleCallbackNextIteration();
+  }
+
+protected:
+  MockReadyCb ready_cb_;
+  Api::ApiPtr api_;
+  DispatcherPtr dispatcher_;
+  std::unique_ptr<Event::UserSpaceFileEventImpl> user_file_event_;
+  Event::SchedulableCallbackPtr io_callback_;
```

Comment:
Maybe this io_callback_ must be destroyed after destroying user_file_event_

Running build asan build on my machine see if I can reproduce.

Reply:
fixed